compatibility/bazel_tools/testing.bzl

```diff
@@ -6,277 +6,174 @@ load(
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.1.0-snapshot.20200506.4107.0.7e448d81": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.1.1": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.2.0-snapshot.20200513.4172.0.021f4af3": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.2.0-snapshot.20200520.4224.0.2af134ca": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "1.2.0-snapshot.20200520.4228.0.595f1e27": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
+    "0.0.0": [
+        {
+            "end": "1.0.1-snapshot.20200417.3908.1.722bac90",
+            "exclusions": ["ContractKeysIT"],
+        },
+        {
+            "start": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "end": "1.1.0-snapshot.20200422.3991.0.6391ee9f",
+            "exclusions": ["ContractKeysIT"],
+        },
+    ],
 }
 
 def get_excluded_tests(test_tool_version, sandbox_version):
-    return excluded_test_tool_tests.get(test_tool_version, default = {}).get(sandbox_version, default = [])
+    exclusion_ranges = excluded_test_tool_tests.get(test_tool_version, default = [])
+    exclusions = []
+    for range in exclusion_ranges:
+        start = range.get("start")
+        end = range.get("end")
+        if start and sandbox_version != "0.0.0" and cmp_version(sandbox_version, start) < 0:
```

Comment:
We're assigning special meaning to the version `0.0.0` by making it always the latest version. I think it would make sense to pull this into `cmp_version`. For versions that don't follow that pattern we can still use [skylib's comparisons](https://github.com/bazelbuild/bazel-skylib/blob/560d7b2359aecb066d81041cb532b82d7354561b/lib/versions.bzl#L57-L80).

Reply:
Makes sense, I’ll make the change. As mentioned below, we cannot use the skylib comparisons since they ignore the prerelease but we can at least use a similar API.

Reply:
Moved to `cmp_version`.